snap/snapcraft.yaml

```diff
@@ -0,0 +1,77 @@
+name: ghostty
+base: core24
+version: git
+summary: A terminal emulator
+description: |
+  Ghostty is a fast, feature-rich, and cross-platform terminal emulator that
+  uses platform-native UI and GPU acceleration.
+grade: stable
+confinement: classic
+contact: https://github.com/ghostty-org/ghostty/discussions
+issues: https://github.com/ghostty-org/ghostty/issues
+website: https://ghostty.org
+license: MIT
+icon: images/icons/icon_512.png
+
+platforms:
+  amd64:
+  arm64:
+
+apps:
+  ghostty:
+    command: bin/ghostty
+    command-chain: [ bin/launcher ]
+    desktop: share/applications/com.mitchellh.ghostty.desktop 
+    environment:
+      PATH: /snap/ghostty/current/bin:/snap/ghostty/current/usr/bin:$PATH
+      LC_ALL: C.UTF-8
+
+parts:
+  launcher:
+    plugin: dump
+    source: snap/local/
+    organize:
+      launcher: bin/
+
+  zig:
+    plugin: nil
+    build-packages:
+      - curl
+    override-pull: |
+      set -ex
+      case "$CRAFT_TARGET_ARCH" in
+        amd64)  arch=x86_64    ;;
+        arm64)  arch=aarch64  ;;
+        *)      arch=""     ;;
+      esac
+
+      rm -rf $CRAFT_PART_SRC/*
+
+      if [[ -n $arch ]]; then
+        curl -LO --retry-connrefused --retry 10 https://ziglang.org/download/0.13.0/zig-linux-$arch-0.13.0.tar.xz
+      fi
+
+      tar xf zig-lin*xz
+      rm -f *xz
+      mv zig-linux*/* .
+    prime:
+      - -*
+
+  ghostty:
+    source: .
+    after: [ zig ]
+    plugin: nil
+    build-packages:
+      - libgtk-4-dev
+      - libadwaita-1-dev
+      - git
+    stage-packages:
```

Comment:
I'm not too familiar with Snap. What are stage packages?

Reply:
stage-packages are ubuntu packages that satisfy runtime dependencies which get bundle in the snap.

Reply:
Hmm, why would we want to bundle shells? Do classic snaps not have access to the host binaries?

Reply:
It does have access to anything on the host, so these could be removed.